Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2332,14 +2334,15 @@ public void applyReferralCode(final String code, final BranchReferralInitListene
      * user selected clients.</p>
      *
      * @param builder A {@link io.branch.referral.Branch.ShareLinkBuilder} instance to build share link.
+     * @return The instance of the {@link Dialog} holding the share view.
      */
-    private void shareLink(ShareLinkBuilder builder) {
+    private Dialog shareLink(ShareLinkBuilder builder) {
```

Comment:
OK The Idea was to give users full handle to the dialog. For e.g We are closing the dialog with an animation. Incase user want to to launch a new activity they want to dismiss the dialog immediately to prevent some  window leaks. With the  Dialog instance they can completely handle the situation.
    We can remove returning the dialog and provide `onDialogLaunched()` and `onDialogDismissed()` callbacks. Does that make sense?